src/user/AddressBookController.ts

```diff
@@ -118,7 +118,7 @@ export class AddressBookController extends BaseController<
    */
   set(address: string, name: string, chainId = '1', memo = '') {
     address = toChecksumAddress(address);
-    if (!isValidAddress(address)) {
+    if (!isValidHexAddress(address)) {
```

Comment:
the line above this where we call toChecksumAddress will fail if dealing with a non 0x prefixed hex string or any other garbage input. I think you could just check if it's a valid Hex address first. Also not a crazy big fan of reassigning a function argument but that's not on you lol.

Reply:
I've added `toChecksumHexAddress` and used it in place of `toChecksumAddress` for safety++